crates/network-scheduler/src/storage.rs

```diff
@@ -358,6 +358,49 @@ impl S3Storage {
             .await
             .map_err(|e| log::error!("Error saving link to assignment: {e:?}"));
         prometheus_metrics::s3_request();
+        self.cleanup_assignments().await;
+    }
+
+    pub async fn cleanup_assignments(&self) {
+        let network = Config::get().network.clone();
+        let path = format!("assignments/{network}/");
+        let Ok(files) = self
+            .client
+            .list_objects()
+            .bucket(&self.config.scheduler_state_bucket)
+            .prefix(path)
+            .send()
+            .await
+        else {
+            log::error!("Failed to list assignments");
+            return;
+        };
+        prometheus_metrics::s3_request();
+        for v in files
+            .contents()
+            .iter()
+            .skip(Config::get().assignment_history_len)
```

Comment:
Wouldn't it remove everything except for the first N files? It should probably be the reverse

Reply:
fixed